pkg/operator/operator.go

```diff
@@ -207,11 +207,14 @@ func ResolveClusterEndpoint(ctx context.Context, eksAPI eksiface.EKSAPI) (string
 	return *out.Cluster.Endpoint, nil
 }
 
-func getCABundle(restConfig *rest.Config) (*string, error) {
+func getCABundle(ctx context.Context, restConfig *rest.Config) (*string, error) {
 	// Discover CA Bundle from the REST client. We could alternatively
 	// have used the simpler client-go InClusterConfig() method.
 	// However, that only works when Karpenter is running as a Pod
 	// within the same cluster it's managing.
+	if caBundle := settings.FromContext(ctx).ClusterCABundle; caBundle != "" {
```

Comment:
If you wanted to support the empty case, you could make it a pointer and have the empty string mean that you want to disable CABundle discovery by default.